hv_networks/J1587Driver.py

```diff
@@ -312,24 +321,30 @@ def make(self):
         return J1708Driver.J1708Driver(self.ports)
 
 
+factory_lock = threading.Lock()
 j1708_factory_singleton = J1708DriverFactory()
 
 
 def set_j1708_driver_factory(factory):
     global j1708_factory_singleton
-    j1708_factory_singleton = factory
+    with factory_lock:
+        j1708_factory_singleton = factory
 
 
 def get_j1708_driver_factory():
-    return j1708_factory_singleton
+    with factory_lock:
+        a = j1708_factory_singleton
+    return a
```

Comment:
Is there a case `a` is not set here?

Reply:
I think the only way that this function doesn't return is if it deadlocks waiting for `factory_lock`, right? Is there something I don't see?

As long as you use the lock responsibly (which this code appears to) it should be fine.

Reply:
Yep. 
Use of the lock is here for publication guarantees of this assignment to the other threads reading the global (also under the lock). 

@salnoobie: `a` is alway (eventually) set.

Reply:
Oh excellent!